packages/@react-aria/color/src/useColorSlider.ts

```diff
@@ -61,6 +62,43 @@ export function useColorSlider(props: ColorSliderAriaOptions, state: ColorSlider
     inputRef
   }, state);
 
+  let {minValue, maxValue, step, pageSize} = state.value.getChannelRange(channel);
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      // these are the cases that useMove or useSlider don't handle
+      if (!/^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+        e.continuePropagation();
+        return;
+      }
+      // same handling as useMove, stopPropagation to prevent useSlider from handling the event as well.
+      e.preventDefault();
+      let channelValue = state.value.getChannelValue(channel);
+      let pageStep = Math.max(pageSize, step);
+      let newValue = channelValue;
+      switch (e.key) {
+        case 'PageUp':
+          newValue = channelValue + pageStep > maxValue ? maxValue : snapValueToStep(channelValue + pageStep, minValue, maxValue, pageStep);
+          break;
+        case 'PageDown':
+          newValue = snapValueToStep(channelValue - pageStep, minValue, maxValue, pageStep);
+          break;
+        case 'Home':
+          newValue = minValue;
+          break;
+        case 'End':
+          newValue = maxValue;
+          break;
+      }
+      // remember to set this so that onChangeEnd is fired
+      state.setThumbDragging(0, true);
+      if (newValue !== channelValue) {
+        state.setValue(state.value.withChannelValue(channel, newValue));
+      }
+      // wait a frame to ensure value has changed then unset this so that onChangeEnd is fired
+      requestAnimationFrame(() => state.setThumbDragging(0, false));
```

Comment:
we should be able to structure this the same way as useSlider https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/slider/src/useSliderThumb.ts#L115
so that we don't need the raf

Reply:
https://github.com/adobe/react-spectrum/pull/2570#discussion_r813453102 might be a concern, lemme double check